### PR TITLE
:speech_balloon: User feedback improvements

### LINKED ISSUE #73: User Feedback Improvements
- [x] Add tooltips to refresh and shuffle buttons
- [x] Make it more obvious that the Submit button is only for when the entire question is done
- [x] Fix the Run button's loading state
- [x] No test cases selected notification for run test cases page
- [x] Fix the Problem Page question skipping issue
- [x] Feedback (notification?) when a highlight is submitted
- [x] Feedback (notification?) when parsons problems are submitted

### PR DESCRIPTION
**Pull request by uoa-ypec413**
## Description

Adds changes to do with improving user feedback on various actions. These include:
- Adds tooltips to refresh, shuffle, edit assignment, log out, and submit buttons.
- Fixes the Run button's loading state
- Adds a notification when no test cases are selected in the run test cases page
- Fixes the Problem Page question skipping issue
- Adds a tick for 300ms when a highlight is submitted
- Adds a tick for 300ms when parsons problems are submitted

Closes #73 

## How has this been tested?

Locally

## Checklist

- [x] I have performed a self-review of my own code
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass

### CODE REVIEW

File: src/components/NavBar/NavBar.tsx [hunk 1]
```diff
@@ -20,12 +20,14 @@ const NavBar = () => {
           <MainLinks />
         </Navbar.Section>
         <Navbar.Section className='justify-self-end mt-auto pb-4 mx-auto'>
-          <GenericButton
-            text='Submit'
-            onClick={openSubmissionModal}
-            disabled={isLoading}
-            loading={isLoading}
-          />
+          <Tooltip label='Submit current question'>
```
Comment (kand198) [inline src/components/NavBar/NavBar.tsx:23]: Nitpick:
Do we want to state that clicking this moves to the next question?
> "Submit question and move to next"

Idk how to word it.

File: src/components/TestCaseSolver/TestCaseSolver.tsx [hunk 2]
```diff
@@ -76,26 +76,30 @@ const TestCaseSolver = () => {
...
-            <IoShuffle
-              size='24px'
-              className={
-                allSolved
-                  ? 'bg-gray-200 stroke-gray-400 rounded-full p-1'
-                  : 'bg-emerald-500 stroke-emerald-50 rounded-full p-1'
-              }
-            />
-          </UnstyledButton>
+          <Tooltip label='Restart tests'>
```
Comment (kand198) [inline src/components/TestCaseSolver/TestCaseSolver.tsx:79]: Nitpick: we refer to them as "test cases" everywhere in the software.
> "Restart test cases"

File: src/components/TestCaseSolver/TestCaseSolver.tsx [hunk 3]
```diff
@@ -76,26 +76,30 @@ const TestCaseSolver = () => {
...
+                size='24px'
+                className={
+                  noneSolved
+                    ? 'bg-gray-200 stroke-gray-400 rounded-full p-1'
+                    : 'bg-emerald-500 stroke-emerald-50 rounded-full p-1'
+                }
+              />
+            </UnstyledButton>
+          </Tooltip>
+          <Tooltip label='Shuffle test'>
```
Comment (kand198) [inline src/components/TestCaseSolver/TestCaseSolver.tsx:91]: Nitpick: I think we probably want to reword to something like
> "Shuffle test cases"

**Review by kand198 — APPROVED** [review]
:+1: Looks great!

File: src/components/NavBar/NavBar.tsx (hunk 1 above)
Comment (uoa-ypec413) [reply src/components/NavBar/NavBar.tsx:23]: yeah

**Comment by kand198** [comment]
Can we change all "submit" buttons to be "save" buttons except for the question submission button?

Also, potentially out of scope: could we have these "save" buttons be deactivated until there are changes to be saved?

**Comment by uoa-ypec413** [comment]
> Can we change all "submit" buttons to be "save" buttons except for the question submission button?
Yes
> Also, potentially out of scope: could we have these "save" buttons be deactivated until there are changes to be saved?
I agree that this is out of scope, can be done in #74

**Review by kand198 — APPROVED** [review]
Awesome, I'm happy with this

**Pull request merged**